### PR TITLE
Add viper cmd functionality and test.

### PR DESCRIPTION
**Pull request by therealmitchconnors**

### CODE REVIEW

File: cmd/cmd.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	fmt.Println("processing viper config")
+	cmd.PersistentFlags().VisitAll(func(f *pflag.Flag) {
+		if reflect.TypeOf(viper.Get(f.Name)).Kind() == reflect.Slice {
+			// Viper cannot convert slices to strings, so this is our workaround.
+			_ = f.Value.Set(strings.Join(viper.GetStringSlice(f.Name), ","))
+		} else {
+			_ = f.Value.Set(viper.GetString(f.Name))
+		}
+	})
+	fmt.Println(viper.GetString("mixerIdentity"))
```
Comment (geeknoid) [inline cmd/cmd.go]: Remove fmt.Printf.

File: cmd/cmd.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+				os.Exit(1)
+			}
+		}
+	})
+}
+
+// ProcessViperConfig retrieves Viper values for each Cobra Val Flag
+func ProcessViperConfig(cmd *cobra.Command, viper *viper.Viper) {
+	viper.SetTypeByDefaultValue(true)
+	fmt.Println("processing viper config")
```
Comment (geeknoid) [inline cmd/cmd.go]: Remove?

**Comment by therealmitchconnors** [comment]
> I'm not happy about the name "cmd" Can this have a more descriptive name?

I'm happy to name it something different, but I'm terrible at names.  This one was just copied from the namespace in istio.  Do you have any recommendations?

**Comment by geeknoid** [comment]
Some ideas:

wrapviper
viperwrap
viperconfig

File: viperconfig/viperconfig_test.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
+// Copyright 2018 Istio Authors
```
Comment (geeknoid) [inline viperconfig/viperconfig_test.go]: Happy New Year!

**Comment by therealmitchconnors** [comment]
^^confirmed flake of monitoring TestSum and TestDistribution^^

**Pull request merged**